### ISSUE TITLE
FxFixture class not exported, but types say it is

### ISSUE THREAD

**Issue by fvsch** [privatenumber/fs-fixture#4]
### Problem

I used `fs-fixture` in a JS project with TypeScript types in JSDoc comments.

I tried declaring a type as:

```js
/** @type {FsFixture} */
let fsFixture;
```

and VS Code inserted an import like this:

```js
import { FsFixture } from 'fs-fixture';
```

which looks correct when looking at `node_modules/fs-fixture/dist/index.d.mts`:

```ts
declare class FsFixture { /* … */ }

export { type FileTree, FsFixture, createFixture };
```

but the built ESM module does not actually export the `FsFixture` class, so I got a runtime error. 

### Expected behavior

Not sure if this is an issue with the output of this library, or with VS Code.

- Could the types be changed to signal that `FsFixture` can only imported as a type, with `export { type FsFixture }` maybe?
- Or would it make sense to expose the `FsFixture` class?

### Minimal reproduction URL

https://stackblitz.com/edit/fs-fixture-class-export-issue?file=index.js

### Version

v2.4.0

### Node.js version

v20.18.0

### Package manager

npm

### Operating system

macOS

### Bugs are expected to be fixed by those affected by it

- [X] I'm interested in working on this issue

### Compensating engineering work financially will speed up resolution

- [ ] I'm willing to offer financial support

**Comment by privatenumber** [comment]
Thanks for the issue.

> Could the types be changed to signal that `FsFixture` can only imported as a type, with `export { type FsFixture }` maybe?

That's how it is.

Source: https://github.com/privatenumber/fs-fixture/blob/b01f20e7fc4a97d574c1538e884555f67e62aa01/src/index.ts#L10

The compiled runtime output ([`index.mjs`](https://unpkg.com/fs-fixture@2.4.0/dist/index.mjs)) doesn't contain `export { type FsFixture }` because that's TS syntax. But the type file, which implies a type export, contains the `FsFixture` export.

**Comment by fvsch** [comment]
> That's how it is.

Not in the compiled `dist/index.d.mts` and `dist/index.d.cts`, which both contain:

```ts
export { type FileTree, FsFixture, createFixture };
```

![Screenshot](https://github.com/user-attachments/assets/52eb114f-90da-4d65-a9b4-bc9956378bf8)

I tried changing that to:

```ts
export { type FileTree, type FsFixture, createFixture };
```

Testing in vscode, I’m getting different auto-generated imports in a `.js` file (ESM):

| `verbatimModuleSyntax` | `fs-fixture/dist/index.d.mts` | Generated import |
| --- | --- | --- |
| `undefined` | `export { FsFixture }` | `import { FsFixture } from 'fs-fixture'` |
| `undefined` | `export { type FsFixture }` | `import { FsFixture } from 'fs-fixture'` |
| `true` | `export { FsFixture }` | `import { FsFixture } from 'fs-fixture'` |
| `true` | `export { type FsFixture }` | `import { type FsFixture } from 'fs-fixture'` |

They're all wrong (correct answer from vscode would be to NOT generate an import at all in the 4th case at least, and maybe in the 2nd case too).

The only thing that seems to work well with vscode in JS files is having types like this:

```diff
- declare class FsFixture {
- interface FsFixture {
```

If you have `declare class Foo` and export `Foo` or even `type Foo`, the TypeScript language server thinks the module exports a `class Foo` (which it doesn't).

**Comment by privatenumber** [comment]
Thanks for the investigation.

> They're all wrong (correct answer from vscode would be to NOT generate an import at all in the 4th case at least, and maybe in the 2nd case too).

I'm not sure if I follow you here. You're saying you reference the `FsFixture` type in your code, but you don't want the type to be imported? How come?


FWIW I'm able to get it to be [`export { type FsFixture }`](https://github.com/privatenumber/fs-fixture/blob/npm/export-fsfixture-type/dist/index.d.mts#L67) in https://github.com/privatenumber/fs-fixture/pull/5

**Comment by fvsch** [comment]
> I'm not sure if I follow you here. You're saying you reference the `FsFixture` type in your code, but you don't want the type to be imported?

So the situation is that I’m writing JavaScript — not TypeScript — that runs in Node.js, with TypeScript types in JSDoc comments. Much like Svelte and other libraries are doing.

At one point in my code I want to use `FsFixture` as a type. Maybe I want to write a custom function that takes a `FsFixture` object as a parameter:

```js
/**
 * @param {FsFixture} fsFixture
 */
function doSomething(fsFixture) {}
```

My problem is a developer usability one, using VS Code. If I just type `@param {FsFixture}`, VS Code shows a dropdown listing the `FsFixture` class, and if I hit Enter or Tab it adds an import for the `FsFixture` class.

So I end up with this code:

```js
import { FsFixture } from 'fs-fixture';

/**
 * @param {FsFixture} fsFixture
 */
function doSomething(fsFixture) {}
```

and if I run this code with Node.js, it crashes:

```
import { FsFixture } from 'fs-fixture';
         ^^^^^^^^^
SyntaxError: The requested module 'fs-fixture' does not provide an export named 'FsFixture'
```

The workaround is easy-ish:

1. Realize that `fs-fixture` does not export a `FsFixture` class. This is maybe the hardest part of the workaround, because VS Code very much says it does!

<img width="621" alt="VS Code screenshot with the code \`import {} from 'fs-fixture'\` and UI listing the following exported names: FsFixture and createFixture" src="https://github.com/user-attachments/assets/5105460e-b32c-4816-99fb-8652feb620f3">

2. Switch to a type import, which with TS-in-JSDoc looks like one of:

```js
/**
 * @param {import('fs-fixture').FsFixture} fsFixture
 */
// or
/**
 * @typedef {import('fs-fixture').FsFixture} FsFixture
 * @param {FsFixture} fsFixture
 */
```

The goal of this issue is to see if the first part can be fixed. That is, make VS Code's TypeScript language service understand that there is no `FsFixture` class.

In my tests, this is only fixed by using `interface FsFixture` instead of `declare class FsFixture`.

Maybe using `declare class FsFixture {}; export { type FsFixture };` (like in #5) should be enough, and that's a TypeScript bug. Or maybe not. I’m not sure about the semantics of `export { type Foo }` in declaration files. (I made a minimal reproduction just in case: https://stackblitz.com/edit/ts-export-type-class?file=test.js)

**Comment by privatenumber** [comment]
I see, I forgot you were using JSDocs.

I just tried it out locally with #5 and it works fine:

https://github.com/user-attachments/assets/d33b6ac7-8ca2-4c07-8a0c-e7bfaca6e2be

**Comment by fvsch** [comment]
Confirmed: fixed in 2.4.1 on my side.
Thanks!
